src/librustc/ty/print/pretty.rs

```diff
@@ -1092,35 +1093,38 @@ impl<F: fmt::Write> Printer<'tcx> for FmtPrinter<'_, 'tcx, F> {
         }
 
         let key = self.tcx.def_key(def_id);
-        if let DefPathData::Impl = key.disambiguated_data.data {
-            // Always use types for non-local impls, where types are always
-            // available, and filename/line-number is mostly uninteresting.
-            let use_types =
-                !def_id.is_local() || {
-                    // Otherwise, use filename/line-number if forced.
-                    let force_no_types = FORCE_IMPL_FILENAME_LINE.with(|f| f.get());
-                    !force_no_types
-                };
+        match key.disambiguated_data.data {
```

Comment:
What's wrong with `if let`?

Reply:
Initially I added another pattern, so `match` made sense... for the sake of generality, I thought it made sense to leave it this way (since I can easily imagine other patterns being added here).

Reply:
I think once you have another pattern to add you should make the change.

Reply:
Fine, okay...